File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/constants/MetadataTypeConstants.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.constants;

import java.util.HashMap;
import java.util.Map;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.content.MetadataDefinitionModel;
import ro.cloudSoft.cloudDoc.presentation.client.shared.providers.GwtLocaleProvider;

public class MetadataTypeConstants {
	
	private static Map<String, String> metadataTypeLabelByTypeName;
	
	public static Map<String, String> getMetadataTypeLabelByTypeName() {
		if (metadataTypeLabelByTypeName == null) {
			metadataTypeLabelByTypeName = new HashMap<String, String>();
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_TEXT, GwtLocaleProvider.getConstants().TEXT());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_NUMERIC, GwtLocaleProvider.getConstants().NUMERIC());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_AUTO_NUMBER, GwtLocaleProvider.getConstants().AUTO_NUMBER());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_DATE, GwtLocaleProvider.getConstants().DATE());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_LIST, GwtLocaleProvider.getConstants().LIST());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_USER, GwtLocaleProvider.getConstants().METADATA_TYPE_USER());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_TEXT_AREA, GwtLocaleProvider.getConstants().EXTENDED_TEXT());
			metadataTypeLabelByTypeName.put(MetadataDefinitionModel.TYPE_METADATA_COLLECTION, GwtLocaleProvider.getConstants().METADATA_COLLECTION());
		}
		return metadataTypeLabelByTypeName;
	}

}

File: CloudFlow/FrontEnd-App/src/app/shared/constants/metadata-type.constants.ts
import { MetadataDefinitionModel } from "./../model/content/metadata-definition.model";

export class MetadataTypeConstants {

	private static metadataTypeLabelCodeByTypeMap: object = null;

	public static getMetadataTypeLabelCodeByTypeMap(): object {
		if (this.metadataTypeLabelCodeByTypeMap === null) {
			this.metadataTypeLabelCodeByTypeMap = {};
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_TEXT] = "TEXT";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_NUMERIC] = "NUMERIC";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_AUTO_NUMBER] = "AUTO_NUMBER";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_DATE] = "DATE";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_DATE_TIME] = "METADATA_TYPE_DATE_TIME";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_MONTH] = "METADATA_TYPE_MONTH";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_LIST] = "LIST";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_USER] = "METADATA_TYPE_USER";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_TEXT_AREA] = "EXTENDED_TEXT";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_METADATA_COLLECTION] = "METADATA_COLLECTION";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_NOMENCLATOR] = "METADATA_TYPE_NOMENCLATOR";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_GROUP] = "METADATA_TYPE_GROUP";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_DOCUMENT] = "METADATA_TYPE_DOCUMENT";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_CALENDAR] = "METADATA_TYPE_CALENDAR";
			this.metadataTypeLabelCodeByTypeMap[MetadataDefinitionModel.TYPE_PROJECT] = "METADATA_TYPE_PROJECT";
		}
		return this.metadataTypeLabelCodeByTypeMap;
	}
}